### PR TITLE
fix(TreeTitle): render selection indicator only when "selectable"

### PR DESCRIPTION
**Pull request by layershifter**
## Current Behavior

`Box` for `selectionIndicator` is rendered always.

## New Behavior

`Box` for `selectionIndicator` is rendered when `Tree` is `selectable`. 

## Solved problem

As we use there `Box` and `.styles` they are not cachable, this causes performance degradation in scrolling for virtual lists.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e86e80a896c035a2b1cb99142dd4ce2e8767942d'>e86e80a896c035a2b1cb99142dd4ce2e8767942d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## Perf Analysis (`@fluentui/react-northstar`)
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
PortalMinimalPerf.default | [186](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/PortalMinimalPerf.default.html) | [171](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.09:1
PopupMinimalPerf.default | [667](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/PopupMinimalPerf.default.html) | [627](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1.06:1
GridMinimalPerf.default | [340](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/GridMinimalPerf.default.html) | [324](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.05:1
HeaderMinimalPerf.default | [361](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/HeaderMinimalPerf.default.html) | [345](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 1.05:1
ReactionMinimalPerf.default | [379](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ReactionMinimalPerf.default.html) | [362](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1.05:1
SkeletonMinimalPerf.default | [344](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SkeletonMinimalPerf.default.html) | [329](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1.05:1
AttachmentMinimalPerf.default | [147](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AttachmentMinimalPerf.default.html) | [141](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 1.04:1
FormMinimalPerf.default | [412](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/FormMinimalPerf.default.html) | [396](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/FormMinimalPerf.default_base.html) | 1.04:1
VideoMinimalPerf.default | [662](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/VideoMinimalPerf.default.html) | [638](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1.04:1
BoxMinimalPerf.default | [343](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/BoxMinimalPerf.default.html) | [334](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/BoxMinimalPerf.default_base.html) | 1.03:1
ButtonMinimalPerf.default | [167](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ButtonMinimalPerf.default.html) | [162](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 1.03:1
CardMinimalPerf.default | [567](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CardMinimalPerf.default.html) | [552](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CardMinimalPerf.default_base.html) | 1.03:1
ListWith60ListItems.default | [621](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListWith60ListItems.default.html) | [603](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListWith60ListItems.default_base.html) | 1.03:1
RefMinimalPerf.default | [239](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/RefMinimalPerf.default.html) | [233](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/RefMinimalPerf.default_base.html) | 1.03:1
SegmentMinimalPerf.default | [338](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SegmentMinimalPerf.default.html) | [329](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 1.03:1
StatusMinimalPerf.default | [684](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/StatusMinimalPerf.default.html) | [666](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.03:1
DividerMinimalPerf.default | [359](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DividerMinimalPerf.default.html) | [353](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.02:1
LoaderMinimalPerf.default | [705](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/LoaderMinimalPerf.default.html) | [692](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 1.02:1
ProviderMergeThemesPerf.default | [1292](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [1270](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 1.02:1
TreeMinimalPerf.default | [825](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TreeMinimalPerf.default.html) | [808](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TreeMinimalPerf.default_base.html) | 1.02:1
AttachmentSlotsPerf.default | [1095](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AttachmentSlotsPerf.default.html) | [1086](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 1.01:1
AvatarMinimalPerf.default | [189](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AvatarMinimalPerf.default.html) | [187](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.01:1
ButtonSlotsPerf.default | [543](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ButtonSlotsPerf.default.html) | [537](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1.01:1
CheckboxMinimalPerf.default | [2765](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CheckboxMinimalPerf.default.html) | [2732](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 1.01:1
DropdownMinimalPerf.default | [3122](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DropdownMinimalPerf.default.html) | [3092](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 1.01:1
EmbedMinimalPerf.default | [4251](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/EmbedMinimalPerf.default.html) | [4189](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 1.01:1
LabelMinimalPerf.default | [387](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/LabelMinimalPerf.default.html) | [383](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/LabelMinimalPerf.default_base.html) | 1.01:1
LayoutMinimalPerf.default | [354](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/LayoutMinimalPerf.default.html) | [350](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.01:1
ListCommonPerf.default | [630](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListCommonPerf.default.html) | [623](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListCommonPerf.default_base.html) | 1.01:1
ListMinimalPerf.default | [515](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListMinimalPerf.default.html) | [511](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListMinimalPerf.default_base.html) | 1.01:1
ProviderMinimalPerf.default | [413](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ProviderMinimalPerf.default.html) | [407](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 1.01:1
RadioGroupMinimalPerf.default | [453](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [448](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.01:1
TableManyItemsPerf.default | [1946](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TableManyItemsPerf.default.html) | [1934](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1.01:1
TextMinimalPerf.default | [339](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TextMinimalPerf.default.html) | [334](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TextMinimalPerf.default_base.html) | 1.01:1
TooltipMinimalPerf.default | [1143](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TooltipMinimalPerf.default.html) | [1131](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1.01:1
AnimationMinimalPerf.default | [554](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AnimationMinimalPerf.default.html) | [552](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 1:1
CarouselMinimalPerf.default | [471](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CarouselMinimalPerf.default.html) | [469](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 1:1
ChatDuplicateMessagesPerf.default | [286](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [287](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 1:1
ChatWithPopoverPerf.default | [395](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ChatWithPopoverPerf.default.html) | [395](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 1:1
DropdownManyItemsPerf.default | [689](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DropdownManyItemsPerf.default.html) | [687](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 1:1
FlexMinimalPerf.default | [278](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/FlexMinimalPerf.default.html) | [277](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/FlexMinimalPerf.default_base.html) | 1:1
HeaderSlotsPerf.default | [766](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/HeaderSlotsPerf.default.html) | [767](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1:1
SliderMinimalPerf.default | [1725](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SliderMinimalPerf.default.html) | [1722](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1:1
SplitButtonMinimalPerf.default | [4428](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [4418](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1:1
IconMinimalPerf.default | [603](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/IconMinimalPerf.default.html) | [602](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/IconMinimalPerf.default_base.html) | 1:1
TextAreaMinimalPerf.default | [485](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TextAreaMinimalPerf.default.html) | [485](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 1:1
CustomToolbarPrototype.default | [2743](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CustomToolbarPrototype.default.html) | [2746](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1:1
ChatMinimalPerf.default | [746](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ChatMinimalPerf.default.html) | [753](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ChatMinimalPerf.default_base.html) | 0.99:1
DialogMinimalPerf.default | [772](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DialogMinimalPerf.default.html) | [778](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DialogMinimalPerf.default_base.html) | 0.99:1
ItemLayoutMinimalPerf.default | [1218](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [1228](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 0.99:1
ListNestedPerf.default | [544](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListNestedPerf.default.html) | [550](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ListNestedPerf.default_base.html) | 0.99:1
MenuButtonMinimalPerf.default | [1707](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1728](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 0.99:1
RosterPerf.default | [1145](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/RosterPerf.default.html) | [1155](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/RosterPerf.default_base.html) | 0.99:1
ButtonOverridesMissPerf.default | [1522](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1546](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 0.98:1
DatepickerMinimalPerf.default | [5750](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DatepickerMinimalPerf.default.html) | [5842](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 0.98:1
InputMinimalPerf.default | [1308](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/InputMinimalPerf.default.html) | [1331](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/InputMinimalPerf.default_base.html) | 0.98:1
MenuMinimalPerf.default | [850](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/MenuMinimalPerf.default.html) | [864](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/MenuMinimalPerf.default_base.html) | 0.98:1
TableMinimalPerf.default | [396](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TableMinimalPerf.default.html) | [405](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TableMinimalPerf.default_base.html) | 0.98:1
ToolbarMinimalPerf.default | [975](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ToolbarMinimalPerf.default.html) | [993](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 0.98:1
ImageMinimalPerf.default | [377](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ImageMinimalPerf.default.html) | [387](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/ImageMinimalPerf.default_base.html) | 0.97:1
AlertMinimalPerf.default | [263](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AlertMinimalPerf.default.html) | [273](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AlertMinimalPerf.default_base.html) | 0.96:1
TreeWith60ListItems.default | [152](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TreeWith60ListItems.default.html) | [160](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/TreeWith60ListItems.default_base.html) | 0.95:1
AccordionMinimalPerf.default | [134](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AccordionMinimalPerf.default.html) | [150](https://fluentuipr.z22.web.core.windows.net/pull/23131/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.89:1

</details><div id="perfComment9424"/>

File: packages/fluentui/react-northstar/src/components/Tree/TreeTitle.tsx [hunk 1]
```diff
@@ -191,17 +191,19 @@ export const TreeTitle = (React.forwardRef<HTMLAnchorElement, TreeTitleProps>((p
...
 
-  const selectIndicator = Box.create(selectionIndicator, {
-    defaultProps: () => ({
-      as: 'span',
-      selected,
-      ...getA11Props('indicator', {
-        className: treeTitleSlotClassNames.indicator,
-        styles: resolvedStyles.selectionIndicator,
+  const selectionIndicatorElement =
+    selectable &&
```
Comment (ling1726) [inline packages/fluentui/react-northstar/src/components/Tree/TreeTitle.tsx:195]: please add a unit test for this

Comment (layershifter) [reply packages/fluentui/react-northstar/src/components/Tree/TreeTitle.tsx:195]: Pushed

**Pull request merged**